internal/controller/bucket/bucket.go

```diff
@@ -342,6 +342,10 @@ func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 	if !ok {
 		return managed.ExternalUpdate{}, errors.New(errNotBucket)
 	}
+	// Fetch the latest version of the bucket to help mitigate "object has been modified" errors.
```

Comment:
@nolancon as i know this also uses the cached client, so maybe still outdate. I think we have to use raw rest client of the current client, or we need a cacheless client. WDYT?